src/driver/linux_resource/syscall_aarch64.c

```diff
@@ -6,15 +6,191 @@
 #include <ci/efrm/debug_linux.h>
 #include <ci/efrm/syscall.h>
 
+#if 1
+#define TRAMP_DEBUG(x...) (void)0
+#else
+#define TRAMP_DEBUG(x...) EFRM_WARN(x)
+#endif
+
 void** efrm_syscall_table = NULL;
 EXPORT_SYMBOL(efrm_syscall_table);
 
 
-static typeof(aarch64_insn_decode_immediate) *aarch64_insn_decode_immediate_sym;
-static typeof(aarch64_insn_read) *aarch64_insn_read_sym;
-static typeof(aarch64_insn_extract_system_reg) *aarch64_insn_extract_system_reg_sym;
-static typeof(aarch64_get_branch_offset) *aarch64_get_branch_offset_sym;
-static typeof(aarch64_insn_adrp_get_offset) *aarch64_insn_adrp_get_offset_sym;
+/* The kernel contains some neat routines for doing AArch64 code inspection.
+ * Some of them are available as inline routines, but some are unfortunately
+ * not exported to modules.
+ * So we use efrm_find_ksym() to resolve these functions if it is available.
+ * Otherwise we copy their implementations here.
+ */
+#ifdef EFRM_HAVE_NEW_KALLSYMS
+static typeof(aarch64_insn_decode_immediate) *ci_aarch64_insn_decode_immediate;
+static typeof(aarch64_insn_read) *ci_aarch64_insn_read;
+static typeof(aarch64_insn_extract_system_reg) *ci_aarch64_insn_extract_system_reg;
+static typeof(aarch64_get_branch_offset) *ci_aarch64_get_branch_offset;
+static typeof(aarch64_insn_adrp_get_offset) *ci_aarch64_insn_adrp_get_offset;
+#else
+/* X-SPDX-Source-URL: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git */
+/* X-SPDX-Source-Tag: v5.15.83 */
+/* X-SPDX-Source-File: arch/arm64/kernel/patching.c */
```

Comment:
Have you considered splitting these into separate files, e.g. 2x .C and 2x .H? For instance, we could add them as-is in one commit and trim them down to what's needed for this PR in the subsequent commit. Perhaps, this could add some clarity?

Given that the patch prefixes the kernel functions with `ci_`, we aren't worrying about the name clashing in newer kernels where these files exist. I suppose if they had the same/unchanged name, we would have to handle three cases: (1) old kernels with `kallsyms_on_each_symbol()` and `EFRM_HAVE_NEW_KALLSYMS`, (2) mid-range, e.g. 5.7..5.13 without neither approach and (3) newer kernels with the patching.c and others. Is it why this PR prefixes them with `ci_`?